lib/Elevate/Components/CCS.pm

```diff
@@ -0,0 +1,388 @@
+use constant TASK_QUEUE_SCRIPT  => '/usr/local/cpanel/bin/servers_queue';
+
+use constant DUMP_TYPES => (
+    calendars => 'ics',
+    contacts  => 'vcard',
+);
+
+sub pre_leapp ($self) {
+    my $ccs_installed = Cpanel::Pkgr::is_installed(CCS_PACKAGE);
+    Elevate::StageFile::update_stage_file( { ccs_installed => $ccs_installed } );
+    return unless $ccs_installed;
+
+    $self->_load_ccs_modules();
+
+    $self->run_once('export_ccs_data');
+    $self->remove_ccs_and_dependencies();
+
+    # Removing the PKG will leave this directory in place
+    # This results in PostGreSQL/CCS failing to start after leapp completes
+    $self->clean_up_pkg_cruft();
+
+    return;
+}
+
+sub clean_up_pkg_cruft ($self) {
+    $self->move_pgsql_directory();
+    $self->remove_cpanel_ccs_home_directory();
+    return;
+}
+
+=head1 remove_cpanel_ccs_home_directory
+
+Removing the package removes the `cpanel-ccs` user, but leaves behind
+the data in the home directory.  This results in the service failing
+to start after the elevation has completed
+
+=cut
+
+sub remove_cpanel_ccs_home_directory ($self) {
+    File::Path::remove_tree('/opt/cpanel-ccs') if -d '/opt/cpanel-ccs';
+    return;
+}
+
+=head1 move_pgsql_directory
+
+Removing the PKG will leave this directory in place
+This results in PostGreSQL/CCS failing to start after leapp completes
+
+=cut
+
+sub move_pgsql_directory ($self) {
+    File::Copy::move( '/var/lib/pgsql', '/var/lib/pgsql_pre_elevate' ) if -d '/var/lib/pgsql';
+    return;
+}
+
+sub remove_ccs_and_dependencies ($self) {
+
+    # There are other dependencies but these are the 3 that we are concerned with
+    my @ccs_depedencies = qw{
+      postgresql
+      postgresql-devel
+      postgresql-server
+    };
+
+    $self->yum->remove( CCS_PACKAGE(), @ccs_depedencies );
+
+    return;
+}
+
+sub _load_ccs_modules ($self) {
+    require Cpanel::LoadModule::Custom;
+
+    Cpanel::LoadModule::Custom::load_perl_module('Cpanel::CCS::Delegates');
+    Cpanel::LoadModule::Custom::load_perl_module('Cpanel::CCS::DBUtils');
+    Cpanel::LoadModule::Custom::load_perl_module('Cpanel::CCS::Userdata');
+
+    return;
+}
+
+=head1
+
+This export code is largely based on the code in
+'/var/cpanel/perl/Cpanel/Pkgacct/Components/CCSPkgAcct.pm' which is provided
+by the CCS package.  Unfortunately, I could not call into that code here
+as that code is intended to be used by '/scripts/pkgacct' and uses
+'Cpanel::Pkgacct::Component' as a parent.  Due to that, I made the decision
+to port that code over to elevate directly.
+
+TL;DR: We will want to monitor CCS for bug fixes/changes as we may need to
+update this code to resolve issues that get resolved there
+
+=cut
+
+sub export_ccs_data ($self) {
+    my $export_dir = EXPORT_DIR();
+    INFO("Exporting CCS data to '$export_dir'.  A backup of this data will be left in place after elevate completes.");
+    $self->_ensure_export_directory();
+
```

Comment:
I'm wondering long this will take on a system with a large number (say a couple thousand) accounts.  Perhaps we should have another INFO line at the end of the function when it is complete.  Also, maybe an INFO statement every 10-100 accounts or so; something like INFO "Exported 200/3500 accounts".  That way, they won't think elevate has hung during this step.

Reply:
Good call.  I'll go ahead and make this change